pkg/config/parameter/file/file_test.go

```diff
@@ -0,0 +1,155 @@
+
+	fileParam := &FileParameter{
+		Path: "myfile",
+	}
+
+	context := parameter.ParameterWriterContext{
+		Parameter: fileParam,
+	}
+
+	result, err := writeFileValueParameter(context)
+	require.NoError(t, err)
+	assert.Equal(t, "myfile", result["path"])
+
+}
+
+func TestWriteFileValueParameter_WrongType(t *testing.T) {
+
+	fileParam := envParam.New("env")
+
+	context := parameter.ParameterWriterContext{
+		Parameter: fileParam,
+	}
+
+	result, err := writeFileValueParameter(context)
+	require.Nil(t, result)
+	assert.IsType(t, &parameter.ParameterWriterError{}, err)
+
+}
+
+func TestParseFileValueParameter_MissingPath(t *testing.T) {
+
+	param, err := parseFileValueParameter(parameter.ParameterParserContext{})
+
+	assert.Nil(t, param)
+	assert.IsType(t, parameter.ParameterParserError{}, err)
+
+}
+
+func TestResolveValue(t *testing.T) {
+	dir, _ := os.MkdirTemp("", "")
+	file, _ := os.CreateTemp(dir, "")
+	file.Write([]byte("test-content"))
+	workingDir, fileName := filepath.Dir(file.Name()), filepath.Base(file.Name())
+
+	param, _ := parseFileValueParameter(parameter.ParameterParserContext{
+		WorkingDir: workingDir,
+		Value:      map[string]any{"path": fileName},
+	})
+
+	result, err := param.ResolveValue(parameter.ResolveContext{})
+	require.NoError(t, err)
+	assert.Equal(t, "test-content", result)
+}
+
+func TestResolveValueWithRefernces(t *testing.T) {
+	dir, _ := os.MkdirTemp("", "")
+	file, _ := os.CreateTemp(dir, "")
+	file.Write([]byte("test-content {{ .ref1 }} - {{ .ref2 }}"))
+	workingDir, fileName := filepath.Dir(file.Name()), filepath.Base(file.Name())
+
+	param, _ := parseFileValueParameter(parameter.ParameterParserContext{
+		WorkingDir: workingDir,
+		Value:      map[string]any{"path": fileName, "references": []any{"ref1", "ref2"}},
+	})
+
+	result, err := param.ResolveValue(parameter.ResolveContext{
+		ResolvedParameterValues: map[string]interface{}{
+			"ref1": "ref1-resolved",
+			"ref2": "ref2-resolved",
+		},
+	})
+	require.NoError(t, err)
+	assert.Equal(t, "test-content ref1-resolved - ref2-resolved", result)
+}
+
+func TestResolveValueWithRefernces_RefMissing(t *testing.T) {
+	dir, _ := os.MkdirTemp("", "")
+	file, _ := os.CreateTemp(dir, "")
+	file.Write([]byte("test-content {{ .ref1 }} - {{ .ref2 }}"))
+	workingDir, fileName := filepath.Dir(file.Name()), filepath.Base(file.Name())
+
+	param, _ := parseFileValueParameter(parameter.ParameterParserContext{
+		WorkingDir: workingDir,
+		Value:      map[string]any{"path": fileName, "references": []any{"ref1", "ref2"}},
+	})
+
+	result, err := param.ResolveValue(parameter.ResolveContext{
+		ResolvedParameterValues: map[string]interface{}{
+			"ref1": "ref1-resolved",
+		},
+	})
+	assert.Nil(t, result)
+	assert.Error(t, err)
+
+}
+
+func TestResolveValue_FileNotFound(t *testing.T) {
+	param, _ := parseFileValueParameter(parameter.ParameterParserContext{
+		WorkingDir: "something",
+		Value:      map[string]any{"path": "something"},
```

Comment:
Little comment: if this is changed to a file the user can't access, the error message gets a bit more cryptic. On a Mac changing `WorkingDir` to `/` and `path` to `.file` produces: `failure trying to render template /.file: failed to read template content: open /.file: permission denied`, meaning it might be worth wrapping the second error in `(f *FileParameter) ResolveValue(...)` in `file.go`, such that the user understands that error happened inside a file parameter.